internal/api/verify.go

```diff
@@ -328,6 +328,19 @@ func (a *API) signupVerify(r *http.Request, ctx context.Context, conn *storage.C
 		if terr = user.Confirm(tx); terr != nil {
 			return internalServerError("Error confirming user").WithInternalError(terr)
 		}
+
+		// on signupVerify, the user will always only have an email identity
+		// so we can safely assume that the first identity is the email identity
+		//
+		// we still check for the length of the identities slice to be safe.
```

Comment:
I like the defensive check on the array bounds and the comment explaining the expected state. Do you think we should also add code that makes the assertions we have in the comment?

Currently it leaves me curious about what should happen when:
 - `len(user.Identities) == 0` when empty should we return an error?
 - `len(user.Identities) > 1` when there is multiple identities should we return an error?

When 1 or more identities is found should we verify the [Identity.Email](https://github.com/supabase/auth/blob/be64d0a270b408abc684a1b5be7428647a7cc4dc/internal/models/identity.go#L27) field matches the Email we have verified? The lack of a check here worries me if a user finds a way to get two unverified identities they could potentially verify one they don't own.

Reply:
> when empty should we return an error?

if it's empty, i think it's fine to not return an error because we might have some old cases from the past before the `auth.identities` table was introduced where a user may not have an identity

> when there is multiple identities should we return an error?

yeah i think it's best to return an error if there are multiple identities because there shouldn't ever be a case where someone signing up has more than 1 identity present

> When 1 or more identities is found should we verify the Identity.Email field matches the Email we have verified?

we can just check if the first identity in the slice has Identity.Provider = "email" and Identity.Email = email